kolibri/plugins/learn/assets/src/views/TopicsPage/index.vue

```diff
@@ -397,18 +398,24 @@
       };
 
       function _handleTopicRedirect(route, children, id, skipped) {
-        if (!children.some(c => !c.is_leaf) && route.name !== PageNames.TOPICS_TOPIC_SEARCH) {
-          // if there are no children which are not leaf nodes (i.e. they have children themselves)
+        if (children.every(c => c.is_leaf) && route.name !== PageNames.TOPICS_TOPIC_SEARCH) {
+          // if all children are leaf nodes (i.e. they have no children themselves)
           // then redirect to search results
-          router.replace({
-            name: PageNames.TOPICS_TOPIC_SEARCH,
-            params: { ...route.params, id },
-            query: route.query,
-          });
+          if (children.every(c => c.title == '')) {
```

Comment:
Could you explain the nested `if` statements here? I understand the first part as the refactor, and I agree that it keeps the condition and makes it easier to read, so thank you. The second one, I'm hoping you could explain why you made this change. Is it to account for empty folders that you mention in the description? If so, I do think having a condition to manage this is a good idea, but I wonder if there might be something a bit easier to logic through than checking by the `c.title`

Reply:
@marcellamaki It does account for the empty folder. I don't exactly remember what I was thinking when I was writing this code(it was long time ago, so I kinda have forgotten it :( ) but, I do remember that I was searching for some difference in router or props that could have made it easier to do. However, they were identical for both the nested and non-nested cases. Therefore, I only found this out so that it automatically redirects whenever all folders are empty. Sorry I wasn't much of help.